tests/common.go

```diff
@@ -6366,3 +6366,12 @@ func GetVolumesFromPoolID(contexts []*scheduler.Context, poolUuid string) ([]*vo
 	}
 	return volumes, nil
 }
+
+// DoRetryWithTimeoutWithGinkgoRecover calls `task.DoRetryWithTimeout` along with `ginkgo.GinkgoRecover()`, to be used in callbacks with panics or ginkgo assertions
+func DoRetryWithTimeoutWithGinkgoRecover(t func() (interface{}, bool, error), timeout, timeBeforeRetry time.Duration) (interface{}, error) {
```

Comment:
IMO, While that makes a short and good name, it's not exactly indicative as to why it is safe and when the "safe" version should be used instead of the regular `task.DoRetryWithTimeout()`. So, this name, despite being long, is clear on **exactly** when and where it should be used and so we should keep it.